### ISSUE TITLE
Question regarding tables discovery for multiple databases

### ISSUE THREAD

**Issue by lizenko** [cavaliercoder/libzbxpgsql#11]
Is there convenient way to discover tables across multiple databases with non-default schema (not public)? Zabbix server configured from scratch and agent configured as described in documentation.
CentOS 7, PostgreSQL version 9.4.4, Zabbix 2.4.6

PostgreSQL databases:
```
bash-4.2$ psql
psql (9.4.4)
Type "help" for help.

postgres=# \c test1
You are now connected to database "test1" as user "postgres".
test1=# set search_path = test1;
SET
test1=# \dt
          List of relations
 Schema |   Name   | Type  |  Owner
--------+----------+-------+----------
 test1  | t1_table | table | postgres
(1 row)

test1=# \d t1_table
    Table "test1.t1_table"
 Column |  Type   | Modifiers
--------+---------+-----------
 i      | integer |
Indexes:
    "t1_table_i_idx" btree (i)

test1=#
```
```
postgres=# \c test2
You are now connected to database "test2" as user "postgres".
test2=# set search_path = test2;
SET
test2=# \dt
          List of relations
 Schema |   Name   | Type  |  Owner
--------+----------+-------+----------
 test2  | t2_table | table | postgres
(1 row)

test2=# \d t2_table
    Table "test2.t2_table"
 Column |  Type   | Modifiers
--------+---------+-----------
 j      | integer |
Indexes:
    "t2_table_j_idx" btree (j)
```

 agents config:
```
[root@sav zabbix]# cat zabbix_agentd.conf | grep -v -e "^$" | grep -v "#"
PidFile=/var/run/zabbix/zabbix_agentd.pid
LogFile=/var/log/zabbix/zabbix_agentd.log
LogFileSize=0
Server=10.0.0.8
ServerActive=10.0.0.8
Hostname=10.0.0.10
Include=/etc/zabbix/zabbix_agentd.d/
[root@sav zabbix]# cat /etc/zabbix/zabbix_agentd.d/libzbxpgsql.conf
LoadModule=libzbxpgsql.so
```

agents log:
```
 18149:20150910:190653.668 Starting Zabbix Agent [10.0.0.10]. Zabbix 2.4.6 (revision 54796).
 18149:20150910:190653.668 using configuration file: /etc/zabbix/zabbix_agentd.conf
 18149:20150910:190653.669 Starting agent module libzbxpgsql 0.2.0
 18149:20150910:190653.669 loaded modules: libzbxpgsql.so
```

Connection string is configured by this way
{$PG_CONN}  -> host=10.0.0.10 port=5432 user=zbx
{$PG_DB} -> postgres

After starting of agent and enabling tables discovery rule I can see successfully established connection to postgres database under zbx user:
```
 2015-09-11 09:50:12 GMT 10.0.0.10 zbx postgres ##LOG:  connection authorized: user=zbx database=postgres
```
then query from pg.table.discovery appears with predictably empty result.
```
 2015-09-11 09:50:12 GMT 10.0.0.10 zbx postgres ##LOG:  duration: 0.257 ms  execute <unnamed>: SELECT     c.oid AS oid     , current_database() AS database     , n.nspn
ame AS schema     , CASE c.reltablespace         WHEN 0 THEN (SELECT ds.spcname FROM pg_tablespace ds JOIN pg_database d ON d.dattablespace = ds.oid WHERE d.datname = c
urrent_database())         ELSE (SELECT spcname FROM pg_tablespace WHERE oid = c.reltablespace)         END AS tablespace     , c.relname AS table     ,t.typname AS typ
e     , pg_catalog.pg_get_userbyid(c.relowner) AS owner     , (SELECT COUNT(inhparent) FROM pg_inherits WHERE inhrelid = c.oid) AS issubclass     , pg_catalog.obj_descr
iption(c.oid, 'pg_class') as description FROM pg_class c JOIN pg_namespace n ON c.relnamespace = n.oid JOIN pg_type t ON c.reltype = t.oid WHERE     c.relkind='r'     A
ND n.nspname <> 'pg_catalog'     AND n.nspname <> 'information_schema'     AND n.nspname !~ '^pg_toast' ORDER BY c.relname
```

If I only change {$PG_DB} to "test1" value in template or set it in host macros - table discovered ok.

Databases discoveries are establishing connections to different databases (not only from {$PG_DB}):
```
 2015-09-11 10:52:24 GMT 10.0.0.10 zbx test1 ##LOG:  duration: 0.195 ms  execute <unnamed>: SELECT SUM(seq_scan::bigint) FROM pg_stat_all_tables WHERE     schemaname <>
 'pg_catalog'     AND schemaname <> 'information_schema'     AND schemaname !~ '^pg_toast'
```
```
 2015-09-11 10:52:24 GMT 10.0.0.10 zbx test2 ##LOG:  duration: 0.220 ms  execute <unnamed>: SELECT SUM(seq_scan::bigint) FROM pg_stat_all_tables WHERE     schemaname <>
 'pg_catalog'     AND schemaname <> 'information_schema'     AND schemaname !~ '^pg_toast'

```

Whats wrong in my configuration? Or how can I discover tables from more than one databases automatically?

**Comment by cavaliercoder** [comment]
Hi, there's nothing wrong with your configuration. Unfortunately one of the limitations of PostgreSQL is that you can only discover tables for the database you are connected to. That's why you only discover tables when you specify the 'test1' database.

I am working on a feature to connect to all accessible databases when doing a table/index/etc. discovery. That should help with this issue, but I'm not really sure how best to implement it.

In the mean time, the answer is to create multiple table discovery rules; one for each database.

I'll leave this open until I've developed a fix.

**Comment by cavaliercoder** [comment]
Fixed from ab7646f, to be released in next version.
Discoveries of tables, schema and indexes will now discover on all connectable databases.
